features/device-config/service/src/main/java/org/opennms/features/deviceconfig/service/impl/DeviceConfigServiceImpl.java

```diff
@@ -179,20 +181,31 @@ private CompletableFuture<PollerResponse> pollDeviceConfig(String ipAddress, Str
 
         final var monitor = getPollerConfig().getServiceMonitor(match.service.getName());
 
-        MonitoredService service = sessionUtils.withReadOnlyTransaction(() -> {
-            OnmsIpInterface ipInterface = ipInterfaceDao.findByIpAddressAndLocation(ipAddress, location);
+        final Pair<Boolean, MonitoredService> boundServicePair = sessionUtils.withReadOnlyTransaction(() -> {
+            final OnmsIpInterface ipInterface = ipInterfaceDao.findByIpAddressAndLocation(ipAddress, location);
             if (ipInterface == null) {
                 return null;
             }
-            OnmsNode node = ipInterface.getNode();
 
-            return new SimpleMonitoredService(ipInterface.getIpAddress(), node.getId(), node.getLabel(), match.serviceName, location);
+            final boolean bound = ipInterface.getMonitoredServices().stream()
+                    .map(OnmsMonitoredService::getServiceName).anyMatch(s -> serviceName.equals(s));
+
+            final OnmsNode node = ipInterface.getNode();
+
+            return Pair.of(bound, new SimpleMonitoredService(ipInterface.getIpAddress(), node.getId(), node.getLabel(), match.serviceName, location));
```

Comment:
may need to add `commons-lang3` feature to `opennms-deviceconfig-service` to get this resolved.

 probably we can use `AbstractMap.SimpleImmutableEntry` to avoid pulling apache commons

Reply:
Good point. Done.